core/analysis/text_token_normalizing_stream.cpp

```diff
@@ -117,85 +118,80 @@ constexpr frozen::unordered_map<
-            auto& case_convert =
-                json[CASE_CONVERT_PARAM_NAME.c_str()];  // optional string enum
+          if (slice.hasKey(CASE_CONVERT_PARAM_NAME)) {
+            auto case_convert_slice = slice.get(CASE_CONVERT_PARAM_NAME);  // optional string enum
 
-            if (!case_convert.IsString()) {
+            if (!case_convert_slice.isString()) {
               IR_FRMT_WARN(
-                  "Non-string value in '%s' while constructing "
-                  "text_token_normalizing_stream from jSON arguments: %s",
-                  CASE_CONVERT_PARAM_NAME.c_str(), args.c_str());
+                "Non-string value in '%s' while constructing "
+                "text_token_normalizing_stream from VPack arguments",
+                CASE_CONVERT_PARAM_NAME.data());
 
               return false;
             }
 
-            auto itr = CASE_CONVERT_MAP.find(case_convert.GetString());
+            auto itr = CASE_CONVERT_MAP.find(irs::get_string<irs::string_ref>(case_convert_slice));
 
             if (itr == CASE_CONVERT_MAP.end()) {
               IR_FRMT_WARN(
-                  "Invalid value in '%s' while constructing "
-                  "text_token_normalizing_stream from jSON arguments: %s",
-                  CASE_CONVERT_PARAM_NAME.c_str(), args.c_str());
+                "Invalid value in '%s' while constructing "
+                "text_token_normalizing_stream from VPack arguments",
+                CASE_CONVERT_PARAM_NAME.data());
 
               return false;
             }
 
             options.case_convert = itr->second;
           }
 
-          if (json.HasMember(ACCENT_PARAM_NAME.c_str())) {
-            auto& accent = json[ACCENT_PARAM_NAME.c_str()];  // optional bool
+          if (slice.hasKey(ACCENT_PARAM_NAME)) {
+            auto accent_slice = slice.get(ACCENT_PARAM_NAME);  // optional bool
 
-            if (!accent.IsBool()) {
+            if (!accent_slice.isBool()) {
               IR_FRMT_WARN(
-                  "Non-boolean value in '%s' while constructing "
-                  "text_token_normalizing_stream from jSON arguments: %s",
-                  ACCENT_PARAM_NAME.c_str(), args.c_str());
+                "Non-boolean value in '%s' while constructing "
+                "text_token_normalizing_stream from VPack arguments",
+                ACCENT_PARAM_NAME.data());
 
               return false;
             }
 
-            options.accent = accent.GetBool();
+            options.accent = accent_slice.getBool();
           }
 
           return true;
         }
       [[fallthrough]];
       default:
         IR_FRMT_ERROR(
-            "Missing '%s' while constructing text_token_normalizing_stream "
-            "from jSON arguments: %s",
-            LOCALE_PARAM_NAME.c_str(), args.c_str());
+          "Missing '%s' while constructing text_token_normalizing_stream "
+          "from VPack arguments",
+          LOCALE_PARAM_NAME.data());
     }
   } catch (...) {
```

Comment:
Let`s add here valocypack exception handling with proper logging. Otherwise we may loose error source.